File: 프로그래머스/unrated/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    point = {}
    for n, y in zip(name, yearning):
        point[n] = y
    
    for p in photo:
        result = 0
        for people in p:
            if people in point:
                result += point[people]
        answer.append(result)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    const yearningMap = new Map();
    for (let i = 0; i < name.length; i++) {
        yearningMap.set(name[i], yearning[i])
    }
    
    const answer = photo.map(people => people.reduce((acc, cur) => {
        return yearningMap.has(cur) ? acc += yearningMap.get(cur) : acc
    }, 0))
    
    
    return answer
}